enable-kdump/ubuntu-enable-kdump.yaml

```diff
@@ -0,0 +1,83 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: enable-kdump
+  labels:
+    app: enable-kdump
+spec:
+  selector:
+    matchLabels:
+      name: enable-kdump
+  template:
+    metadata:
+      labels:
+        name: enable-kdump
+    spec:
+      nodeSelector:
+        "cloud.google.com/gke-os-distribution": "ubuntu"
+        "enable-kdump": "true"
+      hostPID: true
+      containers:
+        - name: enable-kdump
+          image: debian
+          imagePullPolicy: Always
+          securityContext:
+            privileged: true
+          command:
+            - /usr/bin/nsenter
+            - -t 1
+            - -m
+            - -u
+            - -i
+            - -n
+            - -p
+            - --
+            - /bin/bash
+            - -c
+            - |
+
+              set -o errexit
+              set -o pipefail
+              set -o nounset
+
+              function check_kdump() {
+                  local kdump_show
+                  kdump_show=$(kdump-config show)
+                  if echo "${kdump_show}" | grep -q "ready to kdump"; then
+                      echo "ready to kdump!"
+                  else
+                      echo "kdump not setup, isn't ready"
+                  fi
+                  echo "${kdump_show}"
+                  return 0
+              }
+
+              function install() {
+                  echo "installing kdump"
+                  apt-get update
+                  DEBIAN_FRONTEND=noninteractive apt-get install -y kdump-tools
+                  echo "kdump enabled; rebooting..."
+                  sleep 30
```

Comment:
why sleep? maybe turn this down to something smaller like 10?

Reply:
yeah -- what are we waiting for? if it's an active thing, can we have it just inf loop printing the status.

like

Waiting for reboot (x seconds)...
Waiting for reboot (x seconds)...
Waiting for reboot (x seconds)...
Waiting for reboot (x seconds)...
Waiting for reboot (x seconds)...

(NO CARRIER)

Reply:
The sleep is just so we don't do the reboot right away and can ensure we have enough time for the logs from the script to be flushed, etc. 

Added "waiting for reboot", sadly the "NO CARRIER" will have to be client side :)